include/experimental/__p0009_bits/layout_right.hpp

```diff
@@ -68,24 +71,49 @@ struct layout_right {
     // i0+(i1 + E(1)*(i2 + E(2)*i3))
     template <size_t r, size_t Rank>
     struct __rank_count {};
+#if !defined(_MDSPAN_USE_MAPPING_ARG_CAST)
+    // The first option here is safer: it will use the extents size_type and not rely on implicit
+    // promotion of computation of the incoming types. Since everything always gets multiplied
+    // by __extents.extent() there shouldn't be any "unsafe" operations here with respect to
+    // overflowing unless Extents::size_type is not 64bit
 
     template <size_t r, size_t Rank, class I, class... Indices>
-    constexpr size_t __compute_offset(
-      size_t offset, __rank_count<r,Rank>, const I& i, Indices... idx) const {
+    constexpr size_type __compute_offset(
+      size_type offset, __rank_count<r,Rank>, const I& i, Indices... idx) const {
       return __compute_offset(offset * __extents.template __extent<r>() + i,__rank_count<r+1,Rank>(),  idx...);
     }
 
     template<class I, class ... Indices>
-    constexpr size_t __compute_offset(
+    constexpr size_type __compute_offset(
       __rank_count<0,Extents::rank()>, const I& i, Indices... idx) const {
-      return __compute_offset(static_cast<size_t>(i),__rank_count<1,Extents::rank()>(),idx...);
+      return __compute_offset(i,__rank_count<1,Extents::rank()>(),idx...);
     }
 
-    constexpr size_t __compute_offset(size_t offset, __rank_count<Extents::rank(), Extents::rank()>) const {
+#else
+    // This option will cast __extents::extent() to the current index type.
+    // That means if you index into the mapping with just int everything gets
+    // converted to int. That means the user would need to choose index types
```

Comment:
```suggestion
    // converted to int. That means users would need to choose index types
```

Reply:
removed this all, since it was discussion of alternative options, the first ne is what we must DO